### PR TITLE
Docker and README updates

### PR DESCRIPTION
**Pull request by dav3r**
# 

## 🗣 Description

This PR does the following:

- Cleans up the Dockerfile for this project to make it more similar to the [Dockerfile in `skeleton-docker`](https://github.com/cisagov/skeleton-docker/blob/develop/Dockerfile): https://github.com/cisagov/gophish-tools/commit/4597880de0c663cb8315ee0ceb09611943800c70
- Changes most references to `pca` to either `gophish-tools` or `cisa`: https://github.com/cisagov/gophish-tools/commit/368d16edbe42c4cb7ba9318761418bed76a37a8a, https://github.com/cisagov/gophish-tools/commit/2e3c0eb82f1dff0bc89bde8909c5d900d713c6bd
- Adds Docker-related steps to the build workflow: https://github.com/cisagov/gophish-tools/commit/03bc690b89d57a07a8c08b4022f54dad589c9313
- Replaces a homegrown script (`push_readme.sh`) used in the release workflow with a [GitHub action that does the same thing](https://github.com/peter-evans/dockerhub-description): https://github.com/cisagov/gophish-tools/commit/632d1d1ca2835468ffd0e225ae698e3c9a85466a, https://github.com/cisagov/gophish-tools/commit/c563ef32022f7a67cf8cfaad58cbdf779e43b0ad
  @felddy and I discussed whether or not this should be included in [`skeleton-docker`](https://github.com/cisagov/skeleton-docker), but we did not come to a definitive conclusion.  He is making other changes to [`skeleton-docker`](https://github.com/cisagov/skeleton-docker) soon, so we will address it then.
- Comments out a failing test- issue #6 was created to correct this test: https://github.com/cisagov/gophish-tools/commit/264271f37e120ffda4cd4ab9c53a804c459d899e
- Updates and cleans up the README: https://github.com/cisagov/gophish-tools/commit/3525d09c7da5432565497f72857ffe53e4301006, https://github.com/cisagov/gophish-tools/commit/87fc06b591d115299badc0a921febea3ea7ae561

## 💭 Motivation and Context

These changes are necessary to standardize the code in this repo with our other projects that create Docker images.

## 🧪 Testing

I tested these changes locally, as well as in COOL `env1-staging`.

## 🚥 Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (causes existing functionality to change)

## ✅ Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed (uhh, y'know, once I commented out the one that was failing - see #6).

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,25 +1,39 @@
...
-apt-get install --no-install-recommends -y \
-at &&\
-apt-get clean && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
+ARG GIT_COMMIT
+ARG GIT_REMOTE
+ARG VERSION
 
-RUN mkdir ${PCA_HOME} && chown pca:pca ${PCA_HOME}
-VOLUME ${PCA_HOME}
+LABEL git_commit=${GIT_COMMIT}
```
Comment (jsf9k) [inline Dockerfile]: The `{}` are only necessary in cases such as `RUN chmod +x ${GOPHISH_TOOLS_SRC}/var/getenv` where they are necessary to separate the variable name from its surroundings.  You could remove most of them in this file.

File: tests/test_assessment_json.py [hunk 2]
```diff
@@ -1,8 +1,12 @@
 #!/usr/bin/env pytest -vs
 """Tests for assessment JSON."""
 
-# cisagov Libraries
-import assessment.builder as assessment_builder
+# # Standard Python Libraries
```
Comment (jsf9k) [inline tests/test_assessment_json.py:4]: I'm not a fan of leaving commented-out code lying around, but if you're intending to come back to it soon then OK.

**Review by jsf9k — APPROVED** [review]
Approved, but please consider my comments.

File: tests/test_assessment_json.py (hunk 2 above)
Comment (dav3r) [reply tests/test_assessment_json.py:4]: Agreed- @bjb28 is working on fixing this test this week, so I think we will be ok here.

File: Dockerfile (hunk 1 above)
Comment (dav3r) [reply Dockerfile]: All of those curly braces are coming from the [`skeleton-docker` Dockerfile](https://github.com/cisagov/skeleton-docker/blob/develop/Dockerfile), so @felddy, you should include this in your list of changes to make in there.  

In the meantime, I made those changes for this project with https://github.com/cisagov/gophish-tools/pull/7/commits/ce5f17c205b65e0c10df6654713dcb07d39ae710 and verified that I was still able to successfully build the Docker image.  Thanks @jsf9k catching that!

**Pull request merged**